### PR TITLE
DEVICE-23 - Add new properties to `addigy_hostagent`

### PR DESCRIPTION
**Pull request by austinkelleher**
- New properties added to entities:

  | Entity             | Properties                                             |
  | ------------------ | ------------------------------------------------------ |
  | `addigy_hostagent` | `serialNumber`, `wifiMacAddress`, `ethernetMacAddress` |

### CODE REVIEW

File: src/steps/devices/index.ts [hunk 1]
```diff
@@ -55,6 +55,12 @@ export function createHostAgentProtectsHostRelationship(
         model: device['Hardware Model'],
         serial: device['Serial Number'], // This is needed by Device Schema
         serialNumber: device['Serial Number'],
+
+        ethernetMacAddress: agentEntity.ethernetMacAddress as
```
Comment (austinkelleher) [inline src/steps/devices/index.ts:59]: Needed to cast these due to the types in `@jupiterone/integration-sdk-core`.

```ts
declare type EntityPropertyValue = PrimitiveEntityPropertyValue | EntityRawData[];
```

The mapped relationship target entity can't take an `EntityRawData[]` type value...

Comment (i5o) [reply src/steps/devices/index.ts:59]: We are able to add arrays with no issues in _mapping on graph-aws 
image

File: src/steps/devices/index.ts [hunk 2]
```diff
@@ -55,6 +55,12 @@ export function createHostAgentProtectsHostRelationship(
...
+          | string
+          | undefined,
+        wifiMacAddress: agentEntity.ethernetMacAddress as string | undefined,
+        macAddress: agentEntity.macAddress as string[] | undefined,
```
Comment (i5o) [inline src/steps/devices/index.ts:63]:
```suggestion
        wifiMacAddress: agentEntity.ethernetMacAddress,
        macAddress: agentEntity.macAddress
```

File: src/steps/devices/index.ts (hunk 1 above)
Comment (austinkelleher) [reply src/steps/devices/index.ts:59]: The `EntityRawData` property is what is causing the issue. Not arrays.

File: src/steps/devices/converter.ts [hunk 3]
```diff
@@ -26,7 +31,14 @@ export function createHostAgentEntity(device: Device): Entity {
         category: device['Device Model Name'],
         make: 'Apple',
         model: device['Hardware Model'],
-        serial: device['Serial Number'],
+        serial: serialNumber,
+        serialNumber: serialNumber,
```
Comment (i5o) [inline src/steps/devices/converter.ts:35]: nit
```suggestion
       serialNumber,
```

File: src/steps/devices/index.ts (hunk 2 above)
Comment (i5o) [reply src/steps/devices/index.ts:63]: nvm. Just realized

Comment (austinkelleher) [reply src/steps/devices/index.ts:63]: See: https://github.com/JupiterOne/graph-addigy/pull/29#discussion_r1192426016

**Review by i5o — APPROVED** [review]
LGTM

File: src/steps/devices/converter.ts (hunk 3 above)
Comment (austinkelleher) [reply src/steps/devices/converter.ts]: Fixed thanks

**Review by i5o — APPROVED** [review]

**Pull request merged**